### PR TITLE
Fix Dex policy test flake

### PR DESCRIPTION
**Pull request by pasanw**
## Description
After some refactors to policy rendering in #3636, the Dex component inserts rules into rendered policy based on a map iteration. Since map iteration order is random, this was causing flakes when comparing the rendered policy to the expected policy. This PR adjusts the expectation to support differences in ordering.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
